### PR TITLE
fix issue 3790

### PR DESCRIPTION
**Pull request by cristinamanum**
Fix the query performance issue by adding a new index for semverkey.
Add exception to telemetry.
Fix a test issue that is not related with the above change. 

CI build results:
https://nuget.visualstudio.com/NuGetBuild/_build/index?buildId=4703&_a=summary

### CODE REVIEW

File: src/NuGetGallery/Migrations/201704181635170_AddSemVerDeletedIndex.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+namespace NuGetGallery.Migrations
+{
+    using System;
+    using System.Data.Entity.Migrations;
+    
+    public partial class AddSemVerDeletedIndex : DbMigration
+    {
+        public override void Up()
+        {
+            Sql("IF NOT EXISTS(SELECT * FROM sys.indexes WHERE name = 'nci_wi_Packages_SemVerDeleted' AND object_id = OBJECT_ID('Credentials')) CREATE NONCLUSTERED INDEX [nci_wi_Packages_SemVerDeleted] ON [dbo].[Packages]([SemVerLevelKey] ASC,[Deleted] ASC) INCLUDE ([PackageRegistrationKey],[Listed],[Tags],[Title]) ");
```
Comment (joelverhagen) [inline src/NuGetGallery/Migrations/201704181635170_AddSemVerDeletedIndex.cs]: could we make this string multi-line (using `@"..."`) and then format the SQL? this is hard to read

File: src/NuGetGallery/WebApi/QueryResult.cs [hunk 2]
```diff
@@ -92,6 +92,11 @@ public QueryResult(ODataQueryOptions<TModel> queryOptions, IQueryable<TModel> qu
 
                 return response;
             }
+            catch(Exception e)
+            {
+                Telemetry.TrackException(e);
```
Comment (skofman1) [inline src/NuGetGallery/WebApi/QueryResult.cs]: Went over the exception tracking code in Gallery, and QuietLog is the commonly used class for this scenario.

File: src/NuGetGallery/Migrations/201704181635170_AddSemVerDeletedIndex.cs (hunk 1 above)
Comment (joelverhagen) [inline src/NuGetGallery/Migrations/201704181635170_AddSemVerDeletedIndex.cs]: Spoke offline about these things:

- `OBJECT_ID('Credentials')` seems wrong
- Use `Add-Migration` to properly start a new migration
- I want to see the query plan before and after adding the index.

Comment (cristinamanum) [inline src/NuGetGallery/Migrations/201704181635170_AddSemVerDeletedIndex.cs]: Updated.

File: src/NuGetGallery/WebApi/QueryResult.cs (hunk 2 above)
Comment (cristinamanum) [inline src/NuGetGallery/WebApi/QueryResult.cs]: Updated.

**Pull request merged**